src/iframe-helper.js

```diff
@@ -382,15 +386,18 @@ function getSentinel_(iframe, opt_is3P) {
 function parseIfNeeded(data) {
   const shouldBeParsed = typeof data === 'string'
       && data.charAt(0) === '{';
+  let msg;
   if (shouldBeParsed) {
     try {
       data = JSON.parse(data);
     } catch (e) {
       dev().warn('IFRAME-HELPER', 'Postmessage could not be parsed. ' +
           'Is it in a valid JSON format?', e);
     }
+  } else {
+    msg = deserializeMessage(data);
```

Comment:
making that change breaks around 30 tests that I don't have time to fix at the moment